checkmultimemberships.py

```diff
@@ -0,0 +1,138 @@
+###########################################################################################
+# checkmultimemberships --
+#   reads input file [RSU_CACHEFILE]
+#   writes output file [RSU_MULTIFILE], showing current multi-person memberships with only one member
+#
+###########################################################################################
+
+# standard
+import argparse
+from argparse import ArgumentParser
+from csv import DictReader
+from datetime import datetime, timedelta
+#from collections import OrderedDict
+from csv import DictWriter
+
+# pypi
+
+# homegrown
+from running.runsignup import RunSignUp, updatemembercache
+from loutilities.configparser import getitems
+
+from loutilities import timeu
+
+ymd = timeu.asctime('%Y-%m-%d')
+mdy = timeu.asctime('%m/%d/%Y')
+md = timeu.asctime('%m-%d')
+
+import version
+
+class parameterError(Exception): pass
+
+# ----------------------------------------------------------------------
+def analyzemultimembers(membercachefile, multifile=None):
+    # ----------------------------------------------------------------------
+    multimembers = {}
+    multimemberdata = {}
+    multimembersinglecnt = 0
+
+    with open(membercachefile, 'r') as memfile:
+        today = datetime.now()
+        cachedmembers = DictReader(memfile)
+        for memberrec in cachedmembers:
+            enddate = ymd.asc2dt(memberrec['ExpirationDate'])
+
+            if (((memberrec['MembershipType'] == "Family") or
+                (memberrec['MembershipType'] == "Two in Household") or
+                (memberrec['MembershipType'] == "Individual + Junior")) and
+                (enddate >= today)):
+                if ((memberrec['MembershipID']) in multimembers):
+                    multimembers[memberrec['MembershipID']] += 1
+                    if ((memberrec['MembershipID']) in multimemberdata):
+                        del multimemberdata[memberrec['MembershipID']]
+                else:
+                    multimembers[memberrec['MembershipID']] = 1
+                    multimemberdata[memberrec['MembershipID']] = memberrec
+
+    if multifile:
+        with open(multifile, 'w', newline='') as multif:
+            cachehdr = 'MemberID,MembershipID,MembershipType,FamilyName,GivenName,MiddleName,Gender,DOB,Email,PrimaryMember,JoinDate,ExpirationDate,LastModified'.split(
+                ',')
+            cache = DictWriter(multif, cachehdr)

```

Comment:
I understand your use of `cache` is because the code is copied from another script, but this might be better named `report` or similar